src/orange/orangeinp/ConvexRegion.cc

```diff
@@ -309,6 +310,88 @@ void Ellipsoid::output(JsonPimpl* j) const
     to_json_pimpl(j, *this);
 }
 
+//---------------------------------------------------------------------------//
+// GENTRAP
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from half Z height and 1-4 vertices for top and bottom planes
+ */
+GenTrap::GenTrap(real_type halfz, VecReal2 const& lo, VecReal2 const& hi)
+    : hz_{halfz}, lo_{lo}, hi_{hi}
+{
+    CELER_VALIDATE(hz_ > 0, << "nonpositive halfZ: " << hz_);
+    CELER_VALIDATE(lo_.size() <= 4, << "too many vertices for -Z polygon.");
+    CELER_VALIDATE(hi_.size() <= 4, << "too many vertices for +Z polygon.");
+
+    CELER_VALIDATE(lo_.size() >= 3 || hi_.size() >= 3,
+                   << "not enough vertices for both of the +Z/-Z polygons.");
```

Comment:
```suggestion
    CELER_VALIDATE(hz_ > 0, << "nonpositive halfheight: " << hz_);
    CELER_VALIDATE(lo_.size() >= 3 && lo_.size() <= 4, << "invalid number of vertices (" << lo_.size() << ") for -Z polygon");
    CELER_VALIDATE(hi_.size() >= 3 && hi_.size() <= 4, << "invalid number of vertices (" << lo_.size() << ")  for +Z polygon");
```

Reply:
Sizes of 2 or even 1 may still be valid -- as long as `lo_.size() >= 3 || hi_.size() >= 3` still holds.
But I am taking your suggestions for now.